rasa_core/actions/action.py

```diff
@@ -105,3 +107,19 @@ def run(self, dispatcher, tracker, domain):
         if domain.random_template_for("utter_restart") is not None:
             dispatcher.utter_template("utter_restart")
         return [Restarted()]
+
+
+class ActionFallback(Action):
+    """Executes the fallback action and goes back to the previous state
+    of the dialogue"""
+
+    def name(self):
+        return ACTION_FALLBACK_NAME
+
+    def run(self, dispatcher, tracker, domain):
+        from rasa_core.events import UserUtteranceReverted, ActionReverted
+
+        if domain.random_template_for("utter_default") is not None:
+            dispatcher.utter_template("utter_default")
+
+        return [ActionReverted(), UserUtteranceReverted()]
```

Comment:
I don't think we need to revert the action first, if the utterance is reverted all actions after the utterance will be reverted anyways.

Reply:
oh nice, didn't know that